woocommerce/woocommerce-ios
# Introduce `last week`, `month` and `year` ranges

Part of #8146

Why
==========
The final part of issue #8146 is to deliver the `Yesterday`, `Last Week`, `Last Month`, `Last Year`, `Last Quarter` and `Quarter to Date` ranges. This one aims to the `Last Week`, `Last Month`, `Last Year` ranges as the main batch of ranges.

How
==========
This PR introduces the `Last Week`, `Last Month`, `Last Year` ranges, alongside the respective data structs and unit tests, updating the SelectionType and adding the three options to the Date range picker view.

Screen Capture
==========
https://user-images.githubusercontent.com/5920403/205688626-b1b2f254-8919-467e-8aec-5697443fdc44.mp4

How to Test
==========
1. Open the Analytics Hub
2. Hit the Date range selection view
3. Verify that the `Last Week`, `Last Month`, `Last Year` options are now available there. 
4. Verify that the `Last Week` now compares the previous full week to the week before.
5. Verify that the `Last Month` now compares the previous full Month with the month before.
6. Verify that the `Last Year` now compares the last full year (Jan 1 - Dec 31) with the year before it.

Update release notes:
